### PR TITLE
Make winit truly optional

### LINKED ISSUE #64: Separate winit optional feature
With #62 `winit` is now implied when building in "local" mode. We should still make it optional. That may require changes to gfx-rs first, since all of the backends (except `emtpy`) have it enabled by default.

### PR DESCRIPTION
**Pull request by kvark**
Fixes #64

### CODE REVIEW

**Comment by kvark** [comment]
This looks straightforward (and @grovesNL is travelling)
bors r=travis

**Comment by kvark** [comment]
Alright, this is nasty. So when we have a feature that enables "xxx/yyy", where "xxx" is optional, this automatically enables "xxx" out of a sudden. Anybody knows a workaround? @omni-viral hit this before in Amethyst, perhaps?

**Comment by omni-viral** [comment]
I hit it multiple times. There is an issue in cargo (I'd insert link here, but can't find it).
Intermediate solution is listing dependencies that can't be build on any platform under `target.'cfg(...)`.dependencies`
For gfx backends I write it like this:

```toml

gfx-hal = "0.3"
gfx-backend-empty = { version = "0.3", optional = true }
gfx-backend-gl = { version = "0.3", optional = true }

[target.'cfg(all(target_os = "windows", not(target_arch = "wasm32")))'.dependencies]
gfx-backend-dx12 = { version = "0.3", optional = true }

[target.'cfg(all(any(target_os = "windows", all(unix, not(any(target_os = "macos", target_os = "ios")))), not(target_arch = "wasm32")))'.dependencies]
gfx-backend-vulkan = { version = "0.3", optional = true }

[target.'cfg(any(all(target_os = "macos", not(target_arch = "wasm32"), all(target_arch = "aarch64", target_os = "ios"))))'.dependencies]
gfx-backend-metal = { version = "0.3", optional = true }
```

And I have a macro that repeat those `cfg`s for code that uses backend specifics.

**Comment by kvark** [comment]
I decided to go a different route and just route the responsibility to the user of the library (which shouldn't be a problem for wgpu-rs, and possibly easy for others?..). Also filed #7259.

File: wgpu-native/src/instance.rs [hunk 1]
```diff
@@ -218,11 +218,13 @@ pub fn instance_get_adapter(
         let (instance_guard, _) = HUB.instances.read(token);
         instance_guard[instance_id].enumerate_adapters()
     };
-    #[cfg(feature = "gfx-backend-gl")]
+    #[cfg(all(feature = "window-winit", feature = "gfx-backend-gl"))]
     let adapters = {
         let (surface_guard, _) = HUB.surfaces.read(token);
         surface_guard[instance_id].raw.enumerate_adapters()
     };
+    #[cfg(all(not(feature = "windiw-winit"), feature = "gfx-backend-gl"))]
```
Comment (seivan) [inline wgpu-native/src/instance.rs]: typo!

Comment (seivan) [inline wgpu-native/src/instance.rs]:
```suggestion
    #[cfg(all(not(feature = "window-winit"), feature = "gfx-backend-gl"))]
```

**Comment by kvark** [comment]
@seivan thank you for the quick review! Apparently, there was a bunch of code pieces using the old "window-winit" that are now rewritten.

**Comment by kvark** [comment]
Finally, CI appears to be happy.
bors r=seivan

**Comment by omni-viral** [comment]
@kvark This can be real problem because cargo enables features eagerly. User won't be able to enable features based on target platform and thus would end up reexport them same way transitively to the binary. And features would have to be enabled to build final binary.

**Pull request merged**